src/pgmoneta_ext/lib.c

```diff
@@ -52,3 +56,21 @@ pgmoneta_ext_version(PG_FUNCTION_ARGS)
 
    PG_RETURN_DATUM(version);
 }
+
+Datum
+pgmoneta_ext_switch_wal(PG_FUNCTION_ARGS)
+{
+   Datum wal_filename;
+   XLogRecPtr recptr;
+   char* walfile;
+
+   // Call the RequestXLogSwitch function with mark_unimportant as false
+   recptr = RequestXLogSwitch(false);
+
+   // Format the result as a string
+   walfile = psprintf("%X/%X", (uint32) (recptr >> 32), (uint32) recptr);
```

Comment:
I don't know if we should rely this much on postgres' functions. You are risking memory leak here. Perhaps the good old snprintf will suffice.